### PR TITLE
feat: revamp profile pages

### PR DESCRIPTION
**Pull request by lukicenturi**

### CODE REVIEW

**Comment by lukicenturi** [comment]
The features from this PR is done, but unfortunately the `typecheck` is not working. This is the PR to fix it `https://github.com/rotki/rotki.com/pull/145`, but it still need one more thing to do (Check the PR).

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "@nuxt/devtools": "0.8.0",
     "@nuxtjs/i18n": "8.0.0-beta.12",
     "@rotki/eslint-config": "1.1.1",
-    "@rotki/ui-library": "0.3.0",
+    "@rotki/ui-library": "0.4.0",
```
Comment (lukicenturi) [inline package.json:55]: Need to update this again, because https://github.com/rotki/rotki.com/pull/145 still needed.

**Review by tewshi — APPROVED** [review]
lgtm!

File: layouts/account.vue [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  ],
+  ...commonAttrs(),
+});
+
+useAutoLogout();
+const tabModelValue = ref();
+</script>
+
+<template>
+  <Default>
```
Comment (kelsos) [inline layouts/account.vue:33]: is this the only way to use a layout in another layout?

Comment (tewshi) [reply layouts/account.vue:33]: we can use `<NuxtLayout>`

File: components/account/home/PremiumPlaceholder.vue [hunk 3]
```diff
@@ -1,42 +1,39 @@
...
-        <img alt="rotki" src="/img/logo-small.svg" />
-      </div>
-      <div>
-        <TextHeading subheading> Rotki Premium </TextHeading>
-        <div :class="css.premium">
-          No premium subscription found. Select your
-          <NuxtLink :class="css.link" to="/checkout/plan">
-            premium subscription plan.
-          </NuxtLink>
+  <div>
```
[the remark this answers is not in the document]
Comment (tewshi) [reply components/account/home/PremiumPlaceholder.vue:6]: should be fine

File: components/account/home/ApiKeys.vue [hunk 4]
```diff
@@ -8,84 +8,55 @@ const { account } = storeToRefs(store);
...
+        </RuiButton>
       </div>
     </div>
-    <div :class="css.entry">
-      <InputField
-        id="api-secret"
-        :class="css.input"
+    <div class="grid md:grid-cols-2 gap-4">
+      <RuiRevealableTextField
+        id="username"
```
Comment (kelsos) [inline components/account/home/ApiKeys.vue:32]: we have to change the id here and below :)

Comment (tewshi) [reply components/account/home/ApiKeys.vue:32]: on it.

**Comment by kelsos** [comment]
So I think we should not show this this low, and second it should auto-dismiss after some time.

image

**Comment by kelsos** [comment]
Another thing is that there is no visual feedback at all on regenerate of API keys.

The first time it is obvious that something happened but after that there is no indication.

**Comment by kelsos** [comment]
Another thing that seems to be broken is the redirect link to home after buying a subscription.
it points to /home which gives a 404.

File: components/account/home/ApiKeys.vue (hunk 4 above)
Comment (tewshi) [reply components/account/home/ApiKeys.vue]: done

File: components/account/home/PremiumPlaceholder.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (tewshi) [reply components/account/home/PremiumPlaceholder.vue]: done

**Review by kelsos — APPROVED** [review]
looks good to me

**Pull request merged**